src/rust/ensogl/src/display/shape/primitive/def/var.rs

```diff
@@ -411,3 +412,134 @@ define_shape_data_string_operator! { Add add (+) }
 define_shape_data_string_operator! { Sub sub (-) }
 define_shape_data_string_operator! { Mul mul (*) }
 define_shape_data_string_operator! { Div div (/) }
+
+
+
+// ===============================
+// === Trigonometric Functions ===
+// ===============================
+
+impl<T> Sin for Var<T>
+where T: Sin<Output=T> {
+    type Output = Var<T>;
+    fn sin(&self) -> Self {
+        match self {
+            Self::Static  (t) => Var::Static(t.sin()),
+            Self::Dynamic (t) => Var::Dynamic(format!("sin({})",t).into())
+        }
+    }
+}
+
+impl<T> Asin for Var<T>
+where T: Asin<Output=T> {
+    type Output = Var<T>;
+    fn asin(&self) -> Self {
+        match self {
+            Self::Static  (t) => Var::Static(t.asin()),
+            Self::Dynamic (t) => Var::Dynamic(format!("asin({})",t).into())
+        }
+    }
+}
+
+
+impl<T> Cos for Var<T>
+where T: Cos<Output=T> {
+    type Output = Var<T>;
+    fn cos(&self) -> Self {
+        match self {
+            Self::Static  (t) => Var::Static(t.cos()),
+            Self::Dynamic (t) => Var::Dynamic(format!("cos({})",t).into())
+        }
+    }
+}
+
+impl<T> Acos for Var<T>
+    where T: Acos<Output=T> {
+    type Output = Var<T>;
+    fn acos(&self) -> Self {
+        match self {
+            Self::Static  (t) => Var::Static(t.acos()),
+            Self::Dynamic (t) => Var::Dynamic(format!("acos({})",t).into())
+        }
+    }
+}
+
+
+
+// ===================
+// === Square Root ===
+// ===================
+
+impl<T> Sqrt for Var<T>
+where T: Sqrt<Output=T> {
+
+    type Output = Var<T>;
+
+    fn sqrt(&self) -> Self {
+        match self {
+            Self::Static  (t) => Var::Static(t.sqrt()),
+            Self::Dynamic (t) => Var::Dynamic(format!("sqrt({})",t).into())
+        }
+    }
+}
+
+
+
+// ============================
+// === Conversion Functions ===
+// ============================
+// TODO this needs to be revisited with a more generic solution
+
+impl From<Var<Angle<Radians>>> for Var<f32> {
+    fn from(other:Var<Angle<Radians>>) -> Self {
+        match other {
+            Var::Static  (t) => Var::Static(t.value),
+            Var::Dynamic (t) => Var::Dynamic(format!("value({})",t.glsl()).into()),
+        }
+    }
+}
+
+impl From<Var<f32>> for Var<Angle<Radians>> {
+    fn from(other:Var<f32>) -> Self {
+        match other {
+            Var::Static  (t) => Var::Static(Angle::from(t)),
+            Var::Dynamic (t) => Var::Dynamic(format!("Radians({})",t.glsl()).into()),
```

Comment:
This conversion is an internal magic. We have now this magic in two places, which is bad. This should be refactored to a function and this function should be used instead. Otherwise changing the magic in one place is an easy way to make something very bad.

Reply:
I've refactored this code (and the one in `unit.rs`) into helper functions inside of `glrs.rs`